src/utils/Mapper.ts

```diff
@@ -14,18 +14,32 @@ export function toMessage(baseMessage: ITextMessage): Message {
     return returnMessage;
 }
 
-export function toContact(con: IContactRequest | IContactResponse, address: string): Contact {
-    const contact: Contact = {
-        address,
-        name: con.name,
-        myName: '',
-        isActivated: (con as IContactResponse).level !== undefined && (con as IContactResponse).level === Permission.accepted,
-        secret: con.secret,
-        updateTime: con.time,
-        isDisplayed: true,
+export function toContact(con: IContactRequest | IContactResponse, address: string, isGrp: boolean): Contact {
+    if(isGrp) {
+        const contact: Contact = {
+            address,
+            name: con.name,
+            myName: '',
+            isActivated: true,
+            secret: con.secret,
+            updateTime: con.time,
+            isDisplayed: true,
+            isGroup: isGrp,
+        }
+        return contact
+    } else {
+        const contact: Contact = {
+            address,
+            name: con.name,
+            myName: '',
+            isActivated: (con as IContactResponse).level !== undefined && (con as IContactResponse).level === Permission.accepted,
```

Comment:
should be possible to write in much easier way

```suggestion
           isActivated: isGrp ||  (con as IContactResponse).level !== undefined && (con as IContactResponse).level === Permission.accepted,
           
```